src/js/components/stock-movement-wizard/request/AddItemsPage.jsx

```diff
@@ -1407,7 +1407,7 @@ class AddItemsPage extends Component {
           let redirectTo = `/openboxes/stockMovement/list?direction=INBOUND&movementNumber=${movementNumber}&submitted=true`;
 
           if (!this.props.supportedActivities.includes('MANAGE_INVENTORY') && this.props.supportedActivities.includes('SUBMIT_REQUEST')) {
-            redirectTo = '/openboxes/dashboard';
+            redirectTo = `/openboxes/dashboard?requestSubmitted=true&movementNumber=${movementNumber}`;
```

Comment:
Can't you just display the `Alert.success` before redirecting?

Reply:
at the moment there was some kind of a problem with react-router and I couldn't figure it out so  I was redirecting using `window.location` (which refreshes the page and Alert would persist), but now I took another look at it and fixed the issue so the Alert message is displayed before redirect